### PR TITLE
Refractored some parts and added a debug option

### PR DESCRIPTION
**Pull request by BaileyJM02**
Updated a few logic statements to simplify the code and created a `debug` option to ensure that no secret values are knowingly output to the console.

### CODE REVIEW

**Comment by BaileyJM02** [comment]
This pull request has the following changes:
- We now rely on `grep` for pattern matching ([L80-81](https://github.com/jexia/deploy-action/pull/1/files#diff-60254338249f657a0a83f98258a56bfeR80-R81)) to prevent some errors we got before when the `$OUTPUT` was not formatted as expected
- A new error handling process to return more useful outputs when `$STATUS` is not set (i.e the `$OUTPUT` value was not what we expected. ([L83-112](https://github.com/jexia/deploy-action/pull/1/files#diff-60254338249f657a0a83f98258a56bfeR83-R112))
- A new `debug` option for users to include *possibly sensitive* data in the output, see the `README.md` file. This first image is with `debug: false` (default):
image
This second image is with `debug: true`:
image
- A new **internal** `serious_error` function to ignore the `silent_fail` option. ([L11-14](https://github.com/jexia/deploy-action/pull/1/files#diff-60254338249f657a0a83f98258a56bfeR11-R14))
- Better text color handling for readability and maintainability.

---

This will be released in a semantic version `1.1.0` due to new features.

File: deploy.sh [hunk 1]
```diff
@@ -45,54 +47,103 @@ printf "email: $INPUT_EMAIL\npassword: $INPUT_PASSWORD" >>~/.jexia/config.yml
 COMMAND=$(echo "jexia app deploy --project $INPUT_PROJECT_ID --app $INPUT_APP_ID --format shell")
 
 # If the user has set the `wait` input to true we add the `--wait` flag to the command
-if [ $INPUT_WAIT = true ]; then
+if [ "$INPUT_WAIT" = true ]; then
```
Comment (veger) [inline deploy.sh]: `true` is also a string, so best to quote it (although we know it is always there and does not contain spaces, it is best to always quote strings)

```suggestion
if [ "$INPUT_WAIT" = "true" ]; then
```

File: deploy.sh [hunk 2]
```diff
@@ -45,54 +47,103 @@ printf "email: $INPUT_EMAIL\npassword: $INPUT_PASSWORD" >>~/.jexia/config.yml
...
     # Display a warning as a user has a limited amount of minutes and minutes cost money. This should only be used when necessary
-    echo -e "\e[33mWarning: '--wait' flag added. Jexia can take around 7 minutes to deploy. This will wait for the whole duration.\e[39m"
+    echo -e "\e[33mWarning: '--wait' flag added. Jexia can take around 7 minutes to deploy. This will wait for the whole duration.${RESET_COLOR}"
```
Comment (veger) [inline deploy.sh]: Missed replacing the color code to the more readable variables

File: deploy.sh [hunk 3]
```diff
@@ -45,54 +47,103 @@ printf "email: $INPUT_EMAIL\npassword: $INPUT_PASSWORD" >>~/.jexia/config.yml
...
+    *"incorrect email or password"*)
+        # Emulate a known error returned from the deploy command when authentication fails
+        STATUS="Error"
+        ERROR_INFO="unable to authenticate, incorrect email or password provided"
+        serious_error
+        ;;
+    esac
+
+    # If the user has turned debugging on, we will append the whole command response to the end of ERROR_INFO
+    if [ "$INPUT_DEBUG" = true ]; then
```
Comment (veger) [inline deploy.sh]:
```suggestion
    if [ "$INPUT_DEBUG" = "true" ]; then
```

File: deploy.sh [hunk 4]
```diff
@@ -45,54 +47,103 @@ printf "email: $INPUT_EMAIL\npassword: $INPUT_PASSWORD" >>~/.jexia/config.yml
...
+    ;;
+
+"Error")
+    echo -e "${ERROR_COLOR}Deploy failed, ${ERROR_INFO}${RESET_COLOR}"
 
     # This will be useful if they expect to trigger this event frequently where Jexia may not have completed a previous deployment
-    if [ $INPUT_SILENT_FAIL = true ]; then
-        echo -e "\e[31mFailed silently, exit code 0\e[39m"
+    # If it is a serious error, such as the users credentials are incorrect, we will ignore the silent error request.
+    if [ "$INPUT_SILENT_FAIL" = true ] && [ "$SERIOUS_ERROR" = false ]; then
```
Comment (veger) [inline deploy.sh]:
```suggestion
    if [ "${INPUT_SILENT_FAIL}" = "true" ] && [ "${SERIOUS_ERROR}" = "false" ]; then
```

(for consistency reasons I would use the `${variable}` format here as well, like you did in the rest of the script)

**Review by veger — CHANGES_REQUESTED** [review]
You might want to the rest of script for missing quotes around strings, I tried to mark them all, but unchanged lines I did not check.

**Comment by BaileyJM02** [comment]
Hi @veger, thank you for the review, I have ammended the areas you have mentioned. Do you think this is now ready for the `v1.1.0` release?

**Review by veger — APPROVED** [review]

**Comment by veger** [comment]
Looks good to me

**Pull request merged**